ultravox/inference/infer.py

```diff
@@ -33,22 +33,33 @@ def __init__(
     def infer(
         self,
         sample: datasets.VoiceSample,
-        max_tokens: Optional[int] = None,
+        max_new_tokens: Optional[int] = None,
         temperature: Optional[float] = None,
+        past_key_values: Optional[Union[Tuple, transformers.cache_utils.Cache]] = None,
+        num_beams: int = 1,
     ) -> base.VoiceOutput:
         inputs = self._dataproc(sample)
         input_len = inputs["input_ids"].shape[1]
-        output = self._generate(inputs, max_tokens, temperature)
-        output_tokens = output[0][input_len:]
+        output = self._generate(
+            inputs, max_new_tokens, temperature, past_key_values, num_beams
+        )
+        output_tokens = output.sequences[0][input_len:]
```

Comment:
Due to passing `past_key_values` to self._generate(), the type of `output` changes from a list to a GenerateDecoderOnlyOutput. This causes failures in inference unit tests. @juberti Please take a look.

Reply:
FakeInference will probably need to be updated to return a `GenerateDecoderOnlyOutput` as well, should be a straightforward change.